### PR TITLE
[Split Head] Add head which splits location + value prediction

### PR DESCRIPTION
**Pull request by ucinereo**
**New head**:
- Split prediction of value and location
- Value as well as location have a loss cross entropy loss term independent of each other
- Weighted summation of the loss terms

**Refactoring**:
- Instead of passing the transformer to the head, the head is now a property of the transformer.
- Injection of head specific tokens happens at the forward pass. For the original transformer, this happens right at the beginning of the forward pass. For the lens transformer, this happens only after the lens pass.

### CODE REVIEW

File: notebooks/analysis.ipynb
Comment (gserifi) [inline notebooks/analysis.ipynb:1]: Make sure to clear the cell outputs before committing in the future.

File: src/heads/split_action_distribution.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    def __init__(
+        self,
+        n_registers: int,
+        n_action_tokens: int,
+        dim: int,
+        dataset: DatasetBase,
+        alpha: float = 1.0,
+        beta: float = 4.0,
+    ):
+        """Initialize global action distribution head.
```
Comment (gserifi) [inline src/heads/split_action_distribution.py]: Minor detail, but the text here is wrong.

File: src/heads/split_action_distribution.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        loss = self.alpha * v_loss + self.beta * loc_loss
+
+        # Accuracy
+        pred_loc_idx = torch.argmax(loc_pred, dim=-1)  # (b,)
+        pred_v_idx = torch.argmax(v_pred, dim=-1)  # (b,)
+        correct_loc = pred_loc_idx == true_loc_idx
+        correct_val = pred_v_idx == true_v_idx
+        accuracy = (correct_loc & correct_val).float().mean()
+
+        return loss, accuracy
```
Comment (gserifi) [inline src/heads/split_action_distribution.py:174]: It would require some changes to the pipeline, but it could be nice to see the accuracy for both location and value separately. Let's keep that in mind for later.

**Review by gserifi — APPROVED** [review]
Everything looks good! I left some comments, but they are relatively minor. Feel free to either address them now or postpone for later.

File: src/heads/split_action_distribution.py (hunk 1 above)
Comment (gserifi) [reply src/heads/split_action_distribution.py]: Fixed it
